### ISSUE TITLE
[rllib] Access to state space in the exploration object

### ISSUE THREAD

**Issue by DavidMChan** [ray-project/ray#9333]
### What is your question?

Currently, it looks like from the docs, the exploration module (rllib.utils.exploration) only has a way to receive the model-produced logits. Is there a canonical way to access the state as well? We'd like to add some state-based exploration methods (such as count-based exploration) to our models, and it would be great to see this as part of the exploration framework. Similarly, is there any canonical way to train models from within an exploration module (for something like a forward/inverse kinematics model)? 

If this behavior doesn't exist, we'd be happy to work on it for the upstream as well as our own use-case.

**Comment by DavidMChan** [comment]
Bumping this to remove the stale label. .

**Comment by DavidMChan** [comment]
Bumping this to remove the stale label.
